ruby/ql/lib/codeql/ruby/ApiGraphs.qll

```diff
@@ -43,6 +45,16 @@ module API {
      */
     DataFlow::LocalSourceNode getAnImmediateUse() { Impl::use(this, result) }
 
+    /**
+     * Gets a data-flow node corresponding the value flowing into this API component.
+     */
+    DataFlow::Node getARhs() { Impl::def(this, result) }
```

Comment:
Is this `Rhs` as in right-hand side? Right-hand side of what? This predicate seems to be `getInducingNode` specialised to just def nodes, but we don't use it in this PR. Is it necessary? Are there later MaD changes that rely on it maybe?

Reply:
It is necessary for MaD, and you are likely going to use it for future library modelling.  

You can see some examples of how `getARhs()` is used [in my Python PR](https://github.com/github/codeql/pull/7806/commits/e434f075fa00123d2d9104e3ec2d58df250ce538).

Reply:
Ah I see. The name makes sense in the case of `foo.bar = x` as in the example in your PR. For method call parameters it is a bit of a weird name.